std/socket.d

```diff
@@ -1983,6 +2000,8 @@ static if (is(sockaddr_un))
             sun.sun_family = AddressFamily.UNIX;
             sun.sun_path.ptr[0 .. path.length] = (cast(byte[]) path)[];
             sun.sun_path.ptr[path.length] = 0;
+            _nameLen = cast(socklen_t) (sockaddr_un.init.sun_path.offsetof +
+                path.length + (sun.sun_path.ptr[0] ? 1 : 0));
```

Comment:
For pathname addresses, the terminating null is of course not included in `path`, and the manual recommends following these rules:

> *  The pathname in sun_path should be null-terminated.
> *  The length of the pathname, including the terminating null byte, should not exceed the size of sun_path.
> *  The addrlen argument that describes the enclosing sockaddr_un structure should have a value of at least:
> `offsetof(struct sockaddr_un, sun_path)+strlen(addr.sun_path)+1`
> or, more simply, addrlen can be specified as sizeof(struct sockaddr_un).

For abstract addresses, the null byte is the initial one and it is included in the path. The length of the address in this case is `sockaddr_un.init.sun_path.offsetof + path.length`.